lib/authentication/view/sign_up_view.dart

```diff
@@ -27,26 +28,49 @@ class _SignUpViewState extends State<SignUpView> {
   bool agreedToPolicy = false;
 
   final dropdownController = DropdownTreeController();
+  TextEditingController passwordController = TextEditingController();
+  TextEditingController firstNameController = TextEditingController();
+  TextEditingController lastNameController = TextEditingController();
+  TextEditingController emailController = TextEditingController();
+
+  /// Attempt to guess the user's first and last name from the email, since university e-mail addresses are standardized.
+  ///
+  /// Special characters such as ".", "_" are used to separate the names, numbers are removed and names are capitalized.
+  /// *Format example:* firstnameone_firstnametwo.lastname123@stud.acs.pub.ro
+  void parseNameFromEmail(TextEditingController email,
+      TextEditingController firstName, TextEditingController lastName) {
+    String emailWithoutNumbers = email.text.replaceAll(RegExp('[^a-zA-Z._]'), '');
+    List<String> names = emailWithoutNumbers.split('.');
+
+    if (!names[0].contains('_')) {
+      firstName.text = names[0].titleCase;
+    } else {
+      List<String> firstNames = names[0].split('_');
+      firstName.text = firstNames[0].titleCase + ' ' + firstNames[1].titleCase;
```

Comment:
This would only show the first two first names, maybe someone can have three.
```suggestion
      firstName.text = firstNames.map((s) => s.titleCase).join(' ');
```